query_class_config.hh

```diff
@@ -28,14 +28,31 @@ namespace query {
 struct max_result_size {
     uint64_t soft_limit;
     uint64_t hard_limit;
+    uint64_t unlimited_soft_limit;
```

Comment:
We need a comment somewhere on what is this oximoron-sounding "unlimited_soft_limit". It's impossible to guess what's the difference between these four different limits.

Reply:
We certainly don't need 4 limits here, as this is a class internal to the replica. We can do the adaptation logic on the replica entry points.

Reply:
Could you explain @denesb why this is a class internal to the replica? I see it coming from the coordinator through IDL and being constructed on a coordinator side as well.